src/java.base/share/classes/java/util/jar/JarFile.java

```diff
@@ -423,7 +423,9 @@ private Manifest getManifestFromReference() throws IOException {
                     }
                     man = new Manifest(jv, new ByteArrayInputStream(b), getName());
                 } else {
-                    man = new Manifest(super.getInputStream(manEntry), getName());
+                    try (final InputStream is = super.getInputStream(manEntry)) {
+                        man = new Manifest(is, getName());
+                    }
```

Comment:
There is a cleaner so shouldn't have a leak, even if the JarFile is not explicitly closed.
The noisy "final" can be dropped, otherwise looks good.

Reply:
Thank you for the review Alan. I've updated this PR to remove the `final`. And yes as you note, this doesn't really leak. This change closes the InputStream earlier, as soon as it is done, instead of waiting for the `Cleaner` to kick in.